pkg/controller/sparkapplication/controller.go

```diff
@@ -308,201 +299,228 @@ func (c *Controller) syncSparkApplication(key string) error {
 		return err
 	}
 
-	err = c.createSubmission(app)
-	if err != nil {
-		c.recorder.Eventf(
-			app,
-			apiv1.EventTypeWarning,
-			"SparkApplicationSubmissionCreationFailed",
-			"failed to create a submission for SparkApplication %s: %s",
-			app.Name,
-			err.Error())
-		return err
-	}
+	switch app.Status.AppState.State {
+	case "":
+		fallthrough
+	case v1alpha1.NewState:
+		glog.V(2).Infof("Creating Submission for SparkApp: %s", key)
+		err = c.submitSparkApplication(app, 0)
```

Comment:
lets not have side effects from the method. Lets create a deepcopy before we do a submit. then submit can update anything and persistence only happens at the statemanagement layer